geode-core/src/main/java/org/apache/geode/internal/stats50/VMStats50.java

```diff
@@ -621,14 +621,45 @@ public void refresh() {
       }
     }
 
-    refresh(this.heapMemStats, memBean.getHeapMemoryUsage());
-    refresh(this.nonHeapMemStats, memBean.getNonHeapMemoryUsage());
+    refresh(this.heapMemStats, getHeapMemoryUsage(memBean));
+    refresh(this.nonHeapMemStats, getNonHeapMemoryUsage(memBean));
     refreshGC();
     refreshMemoryPools();
     refreshThreads();
   }
 
+  /**
+   * Handle JDK-8207200 gracefully while fetching getHeapMemoryUsage from MemoryMXBean.
+   *
+   * @see <a href="https://bugs.openjdk.java.net/browse/JDK-8207200">JDK-8207200</a>
+   */
+  private MemoryUsage getHeapMemoryUsage(MemoryMXBean memBean) {
+    try {
+      return memBean.getHeapMemoryUsage();
+    } catch (IllegalArgumentException e) {
+      logger.info("JDK-8207200 prevented sampling stats for HeapMemoryUsage");
```

Comment:
I'm concerned that this will log every second (each time a sample is done). Could you add a count somewhere and only log the message once?

Reply:
I had changed it to `debug` but it was in the 2nd commit. I've dropped the 2nd commit and fixed up the 1st commit so you can see it's `debug` now.